### PR TITLE
add Formik for forms 🔥

### PR DESCRIPTION
**Pull request by dereckquock**
- add [Formik](https://jaredpalmer.com/formik/docs/guides/react-native) for `Ask a Question` and `Join Group` forms with [form validation](https://jaredpalmer.com/formik/docs/guides/validation) 🔥 
- add `<TextInput>` wrapper for forms with the form validation
- add stories for the `<TextInput>`, `<AskAboutGroupModal>`, & `<JoinGroupModal>`
- update input styles with a rounded border because [users tend to prefer this for inputs](https://medium.com/google-design/the-evolution-of-material-designs-text-fields-603688b3fe03#1269)

form

### CODE REVIEW

File: components/AskAboutGroupModal.js [hunk 1]
```diff
@@ -32,80 +31,40 @@ function reducer(state, action) {
...
   const handleOpenModal = () => {
     Amplitude.logEventWithProperties('OPEN Group Ask Question', {
-      group: props.title,
+      group: title,
     });
   };
 
-  const handleAsk = () => {
-    if (!firstName || !lastName || !email || !question) {
-      return dropdownAlertRef.current.alertWithType(
```
Comment (dereckquock) [inline components/AskAboutGroupModal.js:75]: Formik includes validation, so the validation error messages are baked into the new `<TextInput>` component

File: components/AskAboutGroupModal.js [hunk 2]
```diff
@@ -125,82 +84,88 @@ export default (props) => {
...
+              <Input
+                autoCompleteType="name"
+                label="First Name"
+                placeholder="Andy"
+                textContentType="givenName"
+                touched={touched.firstName}
+                value={values.firstName}
+                errors={errors.firstName}
+                onChangeText={handleChange('firstName')}
+                onBlur={handleBlur('firstName')}
```
Comment (dereckquock) [inline components/AskAboutGroupModal.js:116]: this stuff if kinda boilerplate, but i think abstracting it is unnecessary

File: components/ModalSheet.js [hunk 3]
```diff
@@ -48,12 +54,15 @@ export default ({
         swipeDirection="down"
         style={styles.modal}
       >
-        <View style={styles.container}>
+        <KeyboardAvoidingView
```
Comment (dereckquock) [inline components/ModalSheet.js:57]: lol...idk how this was working before...
this allows the inputs to get pushed up so the keyboard doesn't cover them

File: components/shared/TextInput.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import { StyleSheet, View, TextInput } from 'react-native';
+import Colors from '../../constants/Colors';
+import { Text } from './Typography';
+
+export default function Input(props) {
+  const { label, touched, errors, ...inputProps } = props;
+
+  return (
+    <View style={styles.container}>
+      {label && (
```
Comment (dereckquock) [inline components/shared/TextInput.js:12]: this is the input's label

File: components/shared/TextInput.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        <Text S bold style={styles.label}>
+          {label}
+        </Text>
+      )}
+      <TextInput
+        autoCorrect={false}
+        keyboardAppearance="dark"
+        returnKeyType="done"
+        placeholderTextColor="rgba(255, 255, 255, 0.5)"
```
Comment (dereckquock) [inline components/shared/TextInput.js:21]: default input props that can be overridden

File: components/shared/TextInput.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        style={styles.input}
+        {...inputProps}
+      />
+      {errors && touched && (
```
Comment (dereckquock) [inline components/shared/TextInput.js]: if the input has an error and it's been visited, then we'll show the error message from our validation function

**Comment by joseph-d-d** [comment]
Approved

**Review by joseph-d-d — DISMISSED** [review]
Approved

**Comment by Phillip3** [comment]
Screen Shot 2020-05-26 at 9 53 33 PM
@dereckquock the background color of the input box is different from the background color of the form (looks like black vs navy). Is it possible to fix that? It immediately stood out as clash-y to me.

File: components/AskAboutGroupModal.js (hunk 2 above)
Comment (Phillip3) [reply components/AskAboutGroupModal.js:116]: For now 😉

File: components/JoinGroupModal.js [hunk 7]
```diff
@@ -31,75 +30,40 @@ function reducer(state, action) {
...
+          showSuccess('Thanks for joining ☺️');
         }
       })
       .catch((err) => {
         dispatch({ type: 'setLoading', value: false });
         Amplitude.logEventWithProperties('ERROR Group Sign Up', {
-          group: props.title,
+          group: title,
           error: err,
         });
```
Comment (Phillip3) [inline components/JoinGroupModal.js:68]: Do you want to show some kind of error message to the user? So they know they didn't actually join the group?

File: components/ModalSheet.js (hunk 3 above)
Comment (Phillip3) [reply components/ModalSheet.js:57]: @dereckquock it wasn't working before. I had to tap on other fields to get it to scroll up for me 😄 I should've said something, but it was a nitpick.

File: ios/Podfile.lock [hunk 8]
```diff
@@ -551,4 +551,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 7f4c21e9567b9edf7b64e41ed7613898a792497c
 
-COCOAPODS: 1.7.5
+COCOAPODS: 1.9.1
```
Comment (Phillip3) [inline ios/Podfile.lock]: @dereckquock won't cocoa pods updates require a new build in the App Store?

**Review by Phillip3 — CHANGES_REQUESTED** [review]
Some Qs and a suggestion.

File: ios/Podfile.lock (hunk 8 above)
Comment (dereckquock) [reply ios/Podfile.lock]: i'm not sure 🤔 but lemme just revert this for now

File: components/AskAboutGroupModal.js (hunk 2 above)
Comment (dereckquock) [reply components/AskAboutGroupModal.js:116]: hmm, i can kinda think of a good way to abstract this...but i'll do it in a separate PR

File: components/JoinGroupModal.js (hunk 7 above)
Comment (dereckquock) [reply components/JoinGroupModal.js:68]: this should basically never happen 🤞 
but i can add some basic error handling in a separate PR

File: components/ModalSheet.js (hunk 3 above)
Comment (dereckquock) [reply components/ModalSheet.js:57]: really? it works for me 🤔

**Review by Phillip3 — APPROVED** [review]
I APPROVE!!!

**Pull request merged**